### PR TITLE
Docs: Add note about removing prepareExecute from the java client

### PR DESCRIPTION
**Pull request by rjernst**
relates #30966

### CODE REVIEW

**Comment by rjernst** [comment]
@elasticmachine run gradle build tests

File: docs/reference/migration/migrate_7_0/java.asciidoc [hunk 1]
```diff
@@ -5,4 +5,10 @@
...
 `isShardsAcked` has been replaced by `isShardsAcknowledged` in
 `CreateIndexResponse`, `RolloverResponse` and
-`CreateIndexClusterStateUpdateResponse`.
\ No newline at end of file
+`CreateIndexClusterStateUpdateResponse`.
+
+==== `prepareExecute` removed from the client api
+
+The `prepareExecute` method which created a request builder has been
+removed from the client api. Instead, construct a build for the
```
Comment (cbuescher) [inline docs/reference/migration/migrate_7_0/java.asciidoc]: not sure if this should be "builder" here

Comment (rjernst) [reply docs/reference/migration/migrate_7_0/java.asciidoc]: The prepareExecute method returned a RequestBuilder. Is there something else you think it should be?

Comment (rjernst) [reply docs/reference/migration/migrate_7_0/java.asciidoc]: Oops, I see what you mean, thanks!

**Review by cbuescher — COMMENTED** [review]
LGTM

**Comment by javanna** [comment]
thanks @rjernst ! This was mainly used by modules and plugins given that for the methods supported by default we still have the prepare* methods around, right? That's what confused me in the original PR, maybe worth mentioning.

**Review by javanna — APPROVED** [review]

**Comment by rjernst** [comment]
I think it would be confusing to mention the other `prepareX` methods, since they are unchanged.

**Comment by javanna** [comment]
I didn't mean to mention unchanged bits, rather to explain clearly who may be impacted by this change (plugins devs rather than users)

**Pull request merged**